lib/text-formatters.js

```diff
@@ -8,8 +8,14 @@ const moment = require('moment');
 moment().format();
 
 function starRating(rating) {
+  const flooredRating =  Math.floor(rating);
   let stars = '';
-  while (stars.length < rating) { stars += '★'; }
+  while (stars.length < flooredRating) { stars += '★'; }
+  const decimal = rating - flooredRating;
+  if (decimal >= 0.875) { stars += '★'; }
+  else if (decimal >= 0.625) { stars += '¾'; }
+  else if (decimal >= 0.375) { stars += '½'; }
+  else if (decimal >= 0.125) { stars += '¼'; }
   while (stars.length < 5) { stars += '☆'; }
   return stars;
 }
```

Comment:
Could you add unit tests of this function, like the ones for `maybePluralize` in https://github.com/badges/shields/blob/master/lib/text-formatters.spec.js? The tests make it easy to verify it's doing what's expected and intended, and prevent regressions in case someone needs to change this in the future.

Reply:
I had missed out on that file. Will fix it.